internal/models/user.go

```diff
@@ -859,3 +841,8 @@ func obfuscatePhone(u *User, phone string) string {
 func obfuscateIdentityProviderId(identity *Identity) string {
 	return obfuscateValue(identity.UserID, identity.Provider+":"+identity.ProviderID)
 }
+
+// FindUserByPhoneChangeAndAudience finds a user with the matching phone change and audience.
+func FindUserByPhoneChangeAndAudience(tx *storage.Connection, phone, aud string) (*User, error) {
+	return findUser(tx, "instance_id = ? and phone_change = ? and aud = ? and is_sso_user = false", uuid.Nil, phone, aud)
```

Comment:
Consider dropping `instance_id` as it looks like it takes a constant value there's no active index that requires it
```suggestion
	return findUser(tx, "phone_change = ? and aud = ? and is_sso_user = false", phone, aud)
```